.github/workflows/default.yml

```diff
@@ -0,0 +1,75 @@
+name: Build Project
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+    name: '[${{ matrix.os }}] Build plugin'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 16
+
+      - name: Cache .gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          gradle-distribution-sha-256-sum-warning: false
+
+      - name: Gradle Wrapper Validation
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: ./gradlew assemble
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: assemble
+
+      - name: ./gradlew check
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: check
+
+      - name: Upload test results
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-results-${{ matrix.os }}
+          path: "${{ github.workspace }}/**/build/reports/tests"
+
+  integration-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+        gradle: [ 6.8.3, current, rc ]
+        java: [ 8, 11, 15 ]
+    name: '[${{ matrix.os }}] Gradle: ${{ matrix.gradle }}, Java: ${{ matrix.java }}'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: set up JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: ${{ matrix.java }}
+
+      - uses: eskatos/gradle-command-action@v1
+        with:
+          build-root-directory: test-project
+          gradle-version: ${{ matrix.gradle }}
+          arguments: formatKotlin lintKotlin
```

Comment:
Tasks called on the test-project are `formatKotlin` and `lintKotlin`. I didn't know how to verify the plugin has been applied correctly, how to verify if it actually works, so I decided to introduce codestyle offences to the test-project and then to run both tasks and see if the command completes. 
Sample scenarios which would result in failing workflow:
- tasks aren't registered propertly
- `formatKotlin` doesn't fix codestyle offences
- any of tasks fail due to dependency conflict (issues like: https://github.com/jeremymailen/kotlinter-gradle/issues/193)
and so on

Sample scenarios that wouldn't be caught:
- `lintKotlin` task becomes no-op
- both tasks become no-op
- custom task, based on i.e. `LintTask` doesn't work

Although, they should fail on regular, TestKit tests